crates/ruma-client-api/src/future/future_message_event.rs

```diff
@@ -0,0 +1,181 @@
+//! `PUT /_matrix/client/*/rooms/{roomId}/send_future/{eventType}/{txnId}`
+//!
+//! Send a future (a scheduled message) to a room. [MSC4140](https://github.com/matrix-org/matrix-spec-proposals/pull/4140)
+
+pub mod v3 {
+    //! `/v3/` ([spec])
+    //!
+    //! [spec]: [MSC4140](https://github.com/matrix-org/matrix-spec-proposals/pull/4140)
+
+    use ruma_common::{
+        api::{request, response, Metadata},
+        metadata,
+        serde::Raw,
+        OwnedRoomId, OwnedTransactionId,
+    };
+    use ruma_events::{AnyMessageLikeEventContent, MessageLikeEventContent, MessageLikeEventType};
+    use serde_json::value::to_raw_value as to_raw_json_value;
+
+    use crate::future::FutureParameters;
+
+    const METADATA: Metadata = metadata! {
+        method: PUT,
+        rate_limited: false,
+        authentication: AccessToken,
+        history: {
+            1.1 => "/_matrix/client/v3/rooms/:room_id/send_future/:event_type/:txn_id",
+        }
+    };
+
+    /// Request type for the `send_future` endpoint.
```

Comment:
the module future is still fine right? since its the module containing multiple endpoints.
`/send_future`
`/state_future`
`/futures` (for updating)

Reply:
What I am saying is that the docs should use the same endpoint name as the names of the ruma modules for the endpoint. So `send_future_message_event`, `send_future_state_event` and `update_future`.

Reply:
got it. I also updated it to actual links.